recipes/cpr/all/conanfile.py

```diff
@@ -101,24 +99,25 @@ def validate(self):
         if not self._uses_valid_abi_and_compiler:
             raise ConanInvalidConfiguration(f"Cannot compile {self.ref} with libstdc++ on clang < 9")
 
-        ssl_library = str(self.options.with_ssl)
-        
-        if ssl_library == "openssl" and is_apple_os(self):
-            raise ConanInvalidConfiguration("OpenSSL is not supported on macOS")
-        if ssl_library == "darwinssl" and not is_apple_os(self):
-            raise ConanInvalidConfiguration("DarwinSSL is only supported on macOS")
-        if ssl_library == "winssl" and self.settings.os != "Windows":
-            raise ConanInvalidConfiguration("WinSSL is only on Windows")
+        if self.options.with_ssl:
+            ssl_library = str(self.options.with_ssl)
 
-        if ssl_library not in (CprConan._AUTO_SSL, CprConan._NO_SSL, "winssl") and ssl_library != self.dependencies["libcurl"].options.with_ssl:
-            raise ConanInvalidConfiguration(
-                f"{self.ref}:with_ssl={self.options.with_ssl} requires libcurl:with_ssl={self.options.with_ssl}"
-            )
+            if ssl_library == "openssl" and is_apple_os(self):
+                raise ConanInvalidConfiguration("OpenSSL is not supported on macOS")
+            if ssl_library == "darwinssl" and not is_apple_os(self):
+                raise ConanInvalidConfiguration("DarwinSSL is only supported on macOS")
+            if ssl_library == "winssl" and self.settings.os != "Windows":
+                raise ConanInvalidConfiguration("WinSSL is only on Windows")
 
-        if ssl_library == "winssl" and self.dependencies["libcurl"].options.with_ssl != "schannel":
-            raise ConanInvalidConfiguration(
-                f"{self.ref}:with_ssl=winssl requires libcurl:with_ssl=schannel"
-            )
+            if ssl_library != "winssl" and ssl_library != self.dependencies["libcurl"].options.with_ssl:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref}:with_ssl={self.options.with_ssl} requires libcurl:with_ssl={self.options.with_ssl}"
+                )
+
+            if ssl_library == "winssl" and self.dependencies["libcurl"].options.with_ssl != "schannel":
+                raise ConanInvalidConfiguration(
+                    f"{self.ref}:with_ssl=winssl requires libcurl:with_ssl=schannel"
+                )
```

Comment:
The problem of this check is that now by default it raises for Windows, because by default cpr:with_ssl=winssl, and libcurl:with_ssl=openssl

So either:
- this check is not correct and should be removed, and ssl backend of libcurl doesn't matter (before this PR, cpr was built with winssl and libcurl with openssl, but it didn't raise because cpr:with_ssl was equal to "auto", and yet there was no issue)
- Change default value of with_ssl of cpr for windows (to openssl), even if it doesn't honor upstream default value

Reply:
I did some testing on Windows, and building with libcurl/*:with_ssl set to OpenSSL and cpr/*:with_ssl set to winssl seems to produce a working binary that can perform https requests. This is the same as the default behavior before, and I will remove the check.